woocommerce/google-listings-and-ads
# Fix Issue disconnecting wpcom api

### Changes proposed in this Pull Request:



Closes # .

This PR addresses an issue when attempting to disconnect all accounts on the settings page. The problem occurs when the user hasn't granted access to the new Pull mechanism, but we still try to revoke the wpcom token. Since there’s no token to revoke, WPCOM responds with an error.

I’m wondering if we should adjust this behavior in WPCOM so that if there’s no token to revoke, it simply returns without throwing an error. I’d be happy to hear thoughts on this.

### Screenshots:




### Detailed test instructions:


1. Check out the develop branch and open the dev console.
2. Navigate to the settings page.
3. Do not grant access to the new API Pull mechanism.
4. Click "Disconnect all accounts."
5. See the error in the dev console from the REST API authorization endpoint, because there is not aWPCOM token to revoke.
6. Refresh the page and go through the onboarding process again.
7. Switch to this branch.
8. Repeat the steps and notice that you can now disconnect all your accounts without any issues.


### Additional details:


### Changelog entry

> Fix - Disconnecting all accounts when WPCOM connection is not granted.
